tests/foreman/cli/test_variables.py

```diff
@@ -752,235 +1145,139 @@ def test_negative_create_matcher_avoid_duplicate(self):
+                'merge-default': 1,
+            })
 
     @run_only_on('sat')
-    @stubbed()
-    @tier3
+    @tier1
     def test_positive_enable_avoid_duplicates_flag(self):
-        """Enable Avoid duplicates flag for supported type- array.
+        """Enable Avoid duplicates flag for supported array type.
 
         @id: 0c7063e0-8e85-44b7-abae-4def7f68833a
 
         @steps:
 
-        1. Set variable type to array.
+        1. Create smart variable with array/hash type.
         2. Set '--merge-overrides' to true.
 
         @assert: The '--avoid-duplicates' flag is allowed to set true.
-
-        @caseautomation: notautomated
-
-        @CaseLevel: System
         """
+        smart_variable = make_variable({
+            'variable': gen_string('alpha'),
+            'puppet-class': self.puppet['name'],
+            'default-value': '[test]',
+            'variable-type': 'array'
+        })
+        SmartVariable.update({
+            'name': smart_variable['name'],
+            'merge-overrides': 1,
+        })
+        SmartVariable.update({
+            'name': smart_variable['name'],
+            'avoid-duplicates': 1,
+        })
+        smart_variable = SmartVariable.info({'id': smart_variable['id']})
+        self.assertEqual(
+            smart_variable['override-values']['avoid-duplicates'], True)
 
     @run_only_on('sat')
-    @stubbed()
-    @tier3
+    @tier1
     def test_negative_enable_avoid_duplicates_flag(self):
-        """Disable Avoid duplicates flag for non supported types.
+        """Attempt to enable Avoid duplicates flag for non supported types.
 
         @id: ea1e8e31-4374-4172-82fd-538d29d70c03
 
         @steps:
 
-        1.  Set variable type other than array.
+        1.  Create smart variable with type that is not array/hash.
+        2.  Attempt to update smart variable and set corresponding flags to
+        True state.
 
         @assert:
 
         1.  The '--merge-overrides' is only allowed to set to true for type
         hash.
         2.  The '--avoid-duplicates' is not allowed to set to true for type
         other than array.
-
-        @caseautomation: notautomated
-
-        @CaseLevel: System
-        """
-
-    @run_only_on('sat')
-    @stubbed()
-    @tier3
-    def test_positive_impact_delete_attribute(self):
-        """Impact on variable after deleting associated attribute.
-
-        @id: ac6f3a65-ed39-4e97-bdee-349f08bd878e
-
-        @steps:
-
-        1.  Create a variable with matcher for some attribute.
-        2.  Delete the attribute.
-        3.  Recreate the attribute with same name as earlier.
-
-        @assert:
-
-        1.  The matcher for deleted attribute removed from variable.
-        2.  On recreating attribute, the matcher should not reappear in
-        variable.
-
-        @caseautomation: notautomated
-
-        @CaseLevel: System
-        """
-
-    @run_only_on('sat')
-    @stubbed()
-    @tier1
-    def test_positive_create_override_from_attribute(self):
```

Comment:
There is no way to interact with smart variables from host interface like we have in UI, so it is invalid case